src/launcher/java/org/truffleruby/launcher/options/CommandLineParser.java

```diff
@@ -432,7 +432,7 @@ private void processArgument() throws CommandLineException {
                         config.setOption(OptionsCatalog.VERBOSITY, "true");
                         break FOR;
                     } else if (argument.startsWith("--dump=")) {
-                        LogWithoutTruffle.LOGGER.warning("the --dump= switch is silently ignored as it is an internal development tool");
+                        TruffleLessLog.LOGGER.warning("the --dump= switch is silently ignored as it is an internal development tool");
```

Comment:
These usages of the Log seem rather simple, we could probably just print to stderr here.
The Logger level itself could be passed as an option to the context.